### DISCUSSION TITLE
Verkürze Gültigkeit von ausgestellten API-Tokens

### DISCUSSION THREAD

**Discussion by MrKrisKrisu** [Traewelling/traewelling#2809] [Ideas / Feature Requests]
Wenn sich ein Benutzer über OAuth authentifiziert, wird ein Token mit einer Gültigkeit von 12 Monaten erstellt. Ich möchte diese Zeit verkürzen, um die Anzahl der im Umlauf befindlichen, gültigen und ungenutzten Token zu reduzieren. Apps können einen aktiv genutzten Token regelmäßig über den [Refresh Endpoint](https://traewelling.de/api/documentation#/Auth/refreshToken) aktualisieren.

Ich bevorzuge eine Gültigkeit von 1 Monat oder weniger. Meinungen?

**Comment by fzakfeld** [comment]
Having worked with some OAuth / OIDC setups I'd go with something very low, 24h max. Refresh tokens exist for a reason ;-)

I generally assume that access tokens could be trusted by resource services without any validity introspection towards the IDP. This means generally access tokens cannot always be revoked. Might be different here, but it still makes sense to follow that idea.

**Reply by MrKrisKrisu** [reply]
24 hours is probably a bit too short when you consider that many tokens are stored locally on user devices (apps). If an app is not opened for 24 hours, nothing can be updated. This can be quite inconvenient for the user, as he needs to log in again every two days...

I would suggest two weeks instead. This is more realistic, as it is unlikely that users will be logged out regularly.

**Comment by fzakfeld** [comment]
For the refresh token, two weeks or more are fine IMO as they can always be revoked.

An access token however can always be fetched with a valid refresh token, even if the user has not been using the app for a while
